geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/fs/storage/parquet/io/SimpleFeatureParquetSchema.scala

```diff
@@ -56,16 +57,64 @@ case class SimpleFeatureParquetSchema(sft: SimpleFeatureType, schema: MessageTyp
 
 object SimpleFeatureParquetSchema {
 
+  import StringSerialization.alphaNumericSafeString
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
   import scala.collection.JavaConverters._
 
   val FeatureIdField = "__fid__"
 
   val SchemaVersionKey = "geomesa.parquet.version"
 
-  val CurrentSchemaVersion = 1
+  val CurrentSchemaVersion = 2
 
-  val GeometryColumnX = "x"
-  val GeometryColumnY = "y"
+  val Encoding = "WKB"
+  val GeoParquetSchemaKey = "geo"
+
+  /**
+   * See https://geoparquet.org/releases/v1.0.0/schema.json
+   *
+   * @param sft simple feature type
+   * @return
+   */
+  def geoParquetMetadata(sft: SimpleFeatureType): String = {
+    val geomField = sft.getGeomField
+
+    // If the sft has no geometry field
+    if (geomField == null) {
+      """{"version":"1.0.0"}"""
+    } else {
+      val primaryColumn = alphaNumericSafeString(geomField)
+      val columns = sft.getAttributeDescriptors.asScala.collect { case g: GeometryDescriptor => geoParquetMetadata(g) }.mkString(",")
+
+      s"""{"version":"1.0.0","primary_column":"$primaryColumn","columns":[$columns"""
```

Comment:
are we missing the end of the array here?
```suggestion
      s"""{"version":"1.0.0","primary_column":"$primaryColumn","columns":[$columns]}"""
```

Reply:
actually, it looks like columns is supposed to be an object, not an array, where the key is the name of the geometry col: https://geoparquet.org/releases/v1.0.0/#:~:text=are%20described%20below.-,File%20metadata,-Field%20Name

Reply:
I intentionally left the array bracket unclosed, because the `finalizeWrite()` method in SimpleFeatureWriteSupport appends the bounding box to the metadata string and closes any open brackets.

Regarding columns needing to be an object: I'm wondering if we should still have the `def geoParquetMetadata(sft: SimpleFeatureType): String` create the entire metadata string directly? Or maybe if it's better to put everything in a struct (or similar data structure) and then marshal it into a json string?

Reply:
Never mind - I added an assert to the `"write parquet files"` unit test in ParquetReadWriteTest that validates the metadata against the json schema, and fixed the json accordingly.